### PR TITLE
Implement OpenAPI documentation generation for procedures

### PR DESCRIPTION
**Pull request by THEjacob1000**
# Pull Request

## Description

- Added a new `describe` method to the `Procedure` class for attaching OpenAPI metadata, including descriptions, summaries, tags, and operation IDs.
- Introduced `OpenAPISpec` and `OpenAPIConfig` interfaces to define the structure of the generated OpenAPI specification.
- Implemented `generateOpenAPISpec` function to create OpenAPI documentation from JSandy routers.
- Enhanced the `Router` class to store operations with their descriptions for OpenAPI generation.
- Added tests to verify the functionality of the new OpenAPI features and ensure proper integration with existing procedures.

## Type of Change

- [x] `feat`: New feature (non-breaking change which adds functionality)
- [ ] `fix`: Bug fix (non-breaking change which fixes an issue)
- [x] `docs`: Documentation changes
- [ ] `style`: Code style changes (formatting, etc.)
- [ ] `refactor`: Code refactoring (no functional changes)
- [x] `test`: Adding or updating tests
- [ ] `chore`: Maintenance tasks
- [ ] `breaking`: Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Affected Packages

- [x] `@jsandy/rpc`
- [ ] `@jsandy/builder`
- [ ] `@jsandy/typescript-config`
- [ ] `create-jsandy-app`
- [x] `www`
- [ ] Root workspace/tooling

## Testing

Please describe how your changes have been tested:

- [x] Unit tests pass (`bun test`)
- [x] Type checking passes (`bun check-types`)
- [x] Linting passes (`bun lint`)
- [x] Build succeeds (`bun run build`)
- [x] Manual testing performed

### Test Coverage

- [x] New features include tests
- [x] Tests cover edge cases
- [x] All tests are passing

### CODE REVIEW

File: packages/rpc/src/router.ts [hunk 1]
```diff
@@ -3,9 +3,13 @@ import { env } from "hono/adapter";
 import { HTTPException } from "hono/http-exception";
 import type { Env, ErrorHandler, MiddlewareHandler } from "hono/types";
 import type { StatusCode } from "hono/utils/http-status";
-import { z, toJSONSchema, ZodObject } from "zod/v4";
+import { ZodObject, z } from "zod/v4";
+import type { JSONSchema } from "zod/v4/core";
 import { bodyParsingMiddleware, queryParsingMiddleware } from "./middleware";
```
[the remark this answers is not in the document]
Comment (THEjacob1000) [reply packages/rpc/src/router.ts:8]: zod/v4/core is not private, it's the stable v4 import until v4 is officially released

File: packages/rpc/src/openapi.ts [hunk 2]
```diff
@@ -0,0 +1,625 @@
...
+	try {
+		// We cannot convert zod v3 schemas to openapi, so we return an empty object
+		if (!("_zod" in zodSchema) && !("$schema" in zodSchema)) {
+			return { type: "object" };
+		}
+		let jsonSchema: JSONSchema.BaseSchema;
+		if (!("$schema" in zodSchema)) {
+			jsonSchema = toJSONSchemaWithDate(zodSchema as ZodType);
+		} else {
+			jsonSchema = zodSchema as JSONSchema.BaseSchema;
```
[the remark this answers is not in the document]
Comment (THEjacob1000) [reply packages/rpc/src/openapi.ts:326]: checking for the _zod type is the documented way of determining if it's a v3 zod object of a v4 zod object
